### PR TITLE
lightning: always use PD HTTP client to access PD HTTP API

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #49865 

Problem Summary:

### What changed and how does it work?

stop using tls.WithHost.GetJSON to access PD HTTP API

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by lance6716** [comment]
lightning CI passed https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fpull_lightning_integration_test/detail/pull_lightning_integration_test/13/pipeline

**Review by D3Hunter — COMMENTED** [review]
15/32 will review later

File: br/pkg/lightning/tikv/tikv.go [hunk 1]
```diff
@@ -253,13 +230,16 @@ func CheckPDVersion(ctx context.Context, tls *common.TLS, pdAddr string,
...
+	pdHTTPCli pdhttp.Client,
+	requiredMinVersion, requiredMaxVersion semver.Version,
+) error {
 	return ForAllStores(
 		ctx,
-		tls.WithHost(pdAddr),
-		StoreStateDown,
-		func(c context.Context, store *Store) error {
+		pdHTTPCli,
+		metapb.StoreState_Offline,
```
Comment (D3Hunter) [inline br/pkg/lightning/tikv/tikv.go:241]: offline == down?

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: br/pkg/lightning/backend/local/tikv_mode.go [hunk 2]
```diff
@@ -63,21 +64,14 @@ func (rc *switcher) switchTiKVMode(ctx context.Context, mode sstpb.SwitchMode, r
 	// since we're running it periodically, so we exclude disconnected stores.
 	// But it is essentially all stores be switched back to Normal mode to allow
 	// normal operation.
-	var minState tikv.StoreState
-	if mode == sstpb.SwitchMode_Import {
-		minState = tikv.StoreStateOffline
-	} else {
-		minState = tikv.StoreStateDisconnected
```
[the remark this answers is not in the document]
Comment (lance6716) [reply br/pkg/lightning/backend/local/tikv_mode.go:70]: These states are no longer exist in current TiKV's code, seems they are v2.0's states. The ground truth of TiKV status is protobuf files.

File: br/pkg/lightning/tikv/tikv.go (hunk 1 above)
Comment (lance6716) [reply br/pkg/lightning/tikv/tikv.go:241]: In the old code's comment, the order is `Tombstone < Offline < Down < Disconnected < Up`. Now we only have `Tombstone >  Offline > Up`, I should have made correct order translation.

**Review by D3Hunter — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by Leavrth — APPROVED** [review]

**Pull request merged**